client/src/main/java/keywhiz/client/KeywhizClient.java

```diff
@@ -272,6 +272,16 @@ public void deleteSecretWithId(long secretId) throws IOException {
     httpDelete(baseUrl.resolve(format("/admin/secrets/%d", secretId)));
   }
 
+  public void deleteSecretWithId(long secretId, String mode) {
+    HttpUrl url = baseUrl.newBuilder()
+        .addPathSegment("admin")
+        .addPathSegment("secrets")
+        .addPathSegment(Long.toString(secretId))
+        .addQueryParameter("mode", mode)
+        .build();
+    httpDeleteQuietly(url);
+  }
```

Comment:
Rather than String.format(), which does no proper escaping whatsoever, we're building up our URL properly.